### DISCUSSION TITLE
Ellipsis position

### DISCUSSION THREAD

**Discussion by wabiloo** [Textualize/rich#3341] [Q&A]
When using overflow "ellipsis", is there a way of defining where the ellipsis should be placed?
Sometimes I need to print strings in which the most significant information is at the end (eg. A path) or both at end and start but not middle (eg. A URL)

**Comment by hetari** [comment]
I think you are talking about `ellipsis` in `Console()`, see this, it may help you:
> [!NOTE] 
> In this code you can remove the `overflow='ellipsis'` and it will show the same output.

```py
from rich.console import Console

console = Console()

long_string = "This is a really really really long string that needs to be truncated"

# Taking the last 30 characters
truncated_start = long_string[-30:]  
console.print(f"Truncated at the start: ...{truncated_start}", overflow='ellipsis')

mid = len(long_string) // 2
truncated_middle = long_string[:15] + "..." + long_string[-15:]
console.print(f"Truncated in the middle: {truncated_middle}", overflow='ellipsis')

truncated_end = long_string[:30]  # Taking the first 30 characters
console.print(f"Truncated at the end: {truncated_end}...", overflow='ellipsis')
```

The output is:
```bash
Truncated at the start: ...ing that needs to be truncated
Truncated in the middle: This is a reall...to be truncated
Truncated at the end: This is a really really really...
```

**Reply by wabiloo** [reply]
OK, you're showing a mechanism where I manage the ellipsis myself, and - as you say - will work in the same way without `overflow='ellipsis'`. I was hoping for a mechanism managed entirely by Rich...

**Reply by hetari** [reply]
I don't think `rich` can do that with the `overflow` option! I searched the docs and saw nothing useful for your case, be sure to search the docs again (maybe you'll find something helpful), and search for other issues too.
